storage.go

```diff
@@ -273,9 +288,27 @@ func (s *Storage) write(ctx context.Context, path string, r io.Reader, size int6
 		r = iowrap.CallbackReader(r, opt.IoCallback)
 	}
 
+	headers := azfile.FileHTTPHeaders{}
+
+	if opt.HasContentType {
+		headers.ContentType = opt.ContentType
+	}
+
+	// `Create` only initializes the file.
+	// ref: https://docs.microsoft.com/en-us/rest/api/storageservices/create-file
+	_, err = s.client.NewFileURL(path).Create(ctx, size, headers, nil)
+	if err != nil {
+		return 0, err
+	}
+
 	body := iowrap.SizedReadSeekCloser(r, size)
+	transactionalMD5, err := base64.StdEncoding.DecodeString(opt.ContentMd5)
```

Comment:
`content_md5` is an optional pair, we should check `HasContentMd5` first.

Reply:
Got it!